systrayhelper: 0.0.4 -> 0.0.5

cryptix opened the pull request:
###### Motivation for this change

updates to the latest release v0.0.5 which includes some fixes and dependency updates.

also includes a checkPhase using xvfb_run to spawn go test, which starts i3 and the helper and then uses xdotool to send clicks back to the helper

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

cryptix committed:
systrayhelper: 0.0.4 -> 0.0.5

nixos-discourse commented:
This pull request has been mentioned on **Nix community**. There might be relevant details there:

https://discourse.nixos.org/t/using-xvfb-in-checkphase/1497/4

rasendubi commented:
Is this a WIP? Otherwise, you need remove commented out code / todos (`don't forget ...`)

cryptix commented:
@rasendubi: indeed an oversight, thanks! 

I'll try to add the ffmpeg record mechanics in pre/postCheckHook so that I can use them through an override from the source.

cryptix commented:
@rasendubi sorry, yes/no. this mostly got dragged behind. I pushed a rebased and cleaned up version.

I moved the commented stuff to export test video recordings into the override.nix in the source of the project to unclutter this one.

Also removed the deps.nix file since this uses vendored dependencies.

rasendubi commented on pkgs/tools/misc/systrayhelper/default.nix:
```diff
@@ -1,21 +1,27 @@
+, dbus
+, i3
+, xdotool
+#, ffmpeg
```
Commented out code?

rasendubi commented on pkgs/tools/misc/systrayhelper/default.nix:
```diff
@@ -29,6 +35,20 @@ buildGoPackage rec {
+  doCheck = stdenv.isLinux;
+  # use the override.nix from the source to extract and watch tests
+  checkPhase = ''
+    export PATH=$PATH:$(pwd)/go/bin:${i3}/bin:${xdotool}/bin
```
Are `i3` and `xdotool` necessary here? I think `checkInputs` should add them to `$PATH` automatically.

rasendubi reviewed (commented):
Minor comments. Otherwise, it looks good.

When you think PR is ready for the final review and merge, remove `[WIP]` from the title.

cryptix commented:
updated to respect review comments and rebased on master.

rasendubi commented:
@GrahamcOfBorg build systrayhelper

rasendubi commented:
nix-review fails for me with the following error:
```
builder for '/nix/store/pbyhaj1ls312xlmvxgxljls5n7m2bwcv-systrayhelper-0.0.5.drv' failed with exit code 1; last 10 log lines:
      xvfb_test.go:32: found i3-msg here: /nix/store/nnskv7cmcv35nvhsac5pcjfddqd6i2lr-i3-4.16.1/bin/i3-msg
      xvfb_test.go:32: found xdotool here: /nix/store/ivkzr71ck6in70p5k8gr0mqm8zbxdi81-xdotool-3.20160805.1/bin/xdotool
      require.go:794: 
          	Error Trace:	xvfb_test.go:31
          	Error:      	Received unexpected error:
          	            	exec: "systrayhelper": executable file not found in $PATH
          	Test:       	TestClicking
          	Messages:   	did not find systrayhelper
  FAIL
  FAIL	github.com/ssbc/systrayhelper/test	0.003s
cannot build derivation '/nix/store/f3fyzpn9y8j0h1kblx5mrhsfhvfb2749-env.drv': 1 dependencies couldn't be built
```

I think `export PATH=$PATH:$(pwd)/go/bin` is still needed in `checkPhase` (without `i3` and `xdotool`)

doronbehar commented:
I'm closing as CI fails and OP seems to not care as for making this update work and to much time has passed.
